Zend/zend_operators.h

```diff
@@ -209,8 +209,16 @@ zend_memnstr(const char *haystack, const char *needle, size_t needle_len, const
 	}
 }
 
+#if defined(HAVE_MEMRCHR) && !defined(__cplusplus) && !defined(i386)
```

Comment:
Why are the `!defined(__cplusplus) && !defined(i386)` conditions necessary?

Reply:
Ah, I saw the explanation in the second commit message. Can you add a comment?

However, for the issue with the prototype and `_GNU_SOURCE`, I think we would be better off if we passed `-D_GNU_SOURCE` to the build (or maybe defined in php_config.h) to make sure it is always defined. At least if it doesn't break something else.

Reply:
``_GNU_SOURCE`` is already defined in `` php_config.h`` but its effect may be different because of different ``#include`` order. ``<string.h>`` may be directly or indirectly  included before `` php_config.h``.

Reply:
In that case I think it would be safest to pass `-D_GNU_SOURCE`, as that should guarantee that it is always defined before the first include (assuming nothing `#undef`s it).

Reply:
agree